app/src/main/java/com/humberto/tasky/core/database/dao/AgendaDao.kt

```diff
@@ -0,0 +1,24 @@
+package com.humberto.tasky.core.database.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import com.humberto.tasky.core.database.entity.EventEntity
+import com.humberto.tasky.core.database.entity.ReminderEntity
+import com.humberto.tasky.core.database.entity.TaskEntity
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface AgendaDao {
+    @Query("SELECT * FROM taskentity WHERE time BETWEEN :startOfDay AND :endOfDay")
+    fun getTasksForDay(startOfDay: Long, endOfDay: Long): Flow<List<TaskEntity>>
+
+    @Query("""
+        SELECT * FROM evententity 
+        WHERE (`from` BETWEEN :startOfDay AND :endOfDay)
+        OR (`to` BETWEEN :startOfDay AND :endOfDay)
+    """)
+    fun getEventsForDay(startOfDay: Long, endOfDay: Long): Flow<List<EventEntity>>
```

Comment:
Wouldn’t checking just the `from` values be enough here to get all the events for a day?

Reply:
This way we wouldn't consider multi-day events, right?

Reply:
It's not precisely true. If an event starts at 23:45 and is scheduled to end at 00:45, then it will also be included in the results. Your first condition will already fetch any event that starts between your `startOfDay` and `endOfDay` regardless of when it is scheduled to end.

Reply:
But what if the event starts Oct 24 and ends on Oct 25 midday for example, and we filter with Oct 25, that event wouldn't be included.

Reply:
🤔 You’ve got a point. If you have an event, for example, that goes over the entire weekend, then you’d want to see it on Friday, Saturday, and Sunday. And that won’t be possible if you just check the start time.

In this case, your query won’t give the event on Saturday because it did not start or end that day. You’d only see it on Friday and Sunday. In this case, feel free to add support for this scenario now or later (if you want to rather focus on getting through the entire app first).

It is excellent that you raised a question like this. It is precisely the type of questions you must raise when you get a requirements document for an app in a real-world project.

Reply:
Awesome, thanks for the discussion guys! 
I'll later make it show every day it covers. :)